lib/standards/html-elms.js

```diff
@@ -0,0 +1,415 @@
+		implicitAttrs: ['aria-level']
+	},
+	head: {
+		allowedRoles: false,
+		noAriaAttrs: true
+	},
+	header: {
+		allowedRoles: ['group', 'none', 'presentation', 'doc-footnote']
+	},
+	hgroup: {
+		allowedRoles: false,
+		noAriaAttrs: true
+	},
+	hr: {
+		allowedRoles: ['none', 'presentation', 'doc-pagebreak']
+	},
+	html: {
+		allowedRoles: false,
+		noAriaAttrs: true
+	},
+	i: {
+		allowedRoles: true
+	},
+	iframe: {
+		allowedRoles: ['application', 'document', 'img', 'none', 'presentation']
+	},
+	img: {
+		features: [
+			{
+				matches: {
+					attributes: {
+						href: '/.+/'
+					}
+				},
+				allowedRoles: [
+					'button',
+					'checkbox',
+					'link',
+					'menuitem',
+					'menuitemcheckbox',
+					'menuitemradio',
+					'option',
+					'progressbar',
+					'scrollbar',
+					'separator',
+					'slider',
+					'switch',
+					'tab',
+					'treeitem',
+					'doc-cover'
+				]
+			},
+			{
+				matches: {
+					attributes: {
+						href: ''
+					}
+				},
+				allowedRoles: ['presentation', 'none']
+			},
+			{
+				matches: {
+					attributes: {
+						href: null
+					}
+				},
+				// note: if img has accessible name then uses alt="some text"
+				// but we can't match that so we'll go with false
+				allowedRoles: false
+			}
+		]
+	},
+	input: {
+		features: [
+			{
+				matches: {
+					attributes: {
+						type: 'button'
+					}
+				},
+				allowedRoles: [
+					'link',
+					'menuitem',
+					'menuitemcheckbox',
+					'menuitemradio',
+					'option',
+					'radio',
+					'switch',
+					'tab'
+				]
+			},
+			{
+				matches: {
+					attributes: {
+						type: 'checkbox',
+						'aria-pressed': '/.*/'
+					}
+				},
+				allowedRoles: ['button', 'menuitemcheckbox', 'option', 'switch'],
+				implicitAttrs: ['aria-checked']
+			},
+			{
+				matches: {
+					attributes: {
+						type: 'checkbox',
+						'aria-pressed': null
+					}
+				},
+				allowedRoles: ['menuitemcheckbox', 'option', 'switch'],
+				implicitAttrs: ['aria-checked']
+			},
+			{
+				matches: {
+					attributes: {
+						type: [
+							'color',
+							'date',
+							'datetime-local',
+							'email',
+							'file',
+							'month',
+							'number',
+							'password',
+							'range',
+							'reset',
+							'search',
+							'submit',
+							'tel',
+							'time',
+							'url',
+							'week'
+						]
+					}
+				},
+				allowedRoles: false
+			},
+			{
+				matches: {
+					attributes: {
+						type: 'hidden'
+					}
+				},
+				allowedRoles: false,
+				noAriaAttrs: true
+			},
+			{
+				matches: {
+					attributes: {
+						type: 'image'
+					}
+				},
+				allowedRoles: [
+					'link',
+					'menuitem',
+					'menuitemcheckbox',
+					'menuitemradio',
+					'radio',
+					'switch'
+				]
+			},
+			{
+				matches: {
+					attributes: {
+						type: 'radio'
+					}
+				},
+				allowedRoles: ['menuitemradio'],
+				implicitAttrs: ['aria-checked']
+			},
+			{
+				matches: {
+					attributes: {
+						type: 'text',
```

Comment:
This is a bit of a problem... something with an invalid type, or without a type will default to type text. That's not captured with this logic.